pkg/health/health.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+)
+
+func Health(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	fmt.Println("Info:  Checking if at least three mon pods are running on different nodes")
+	checkPodsOnNodes(context, operatorNamespace, clusterNamespace, "app=rook-ceph-mon", exitOnError)
+
+	fmt.Println()
+	fmt.Println(" Info:  Checking mon quorum and ceph health details")
+	checkMonQuorum(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  Checking if at least three osd pods are running on different nodes")
+	checkPodsOnNodes(context, operatorNamespace, clusterNamespace, "app=rook-ceph-osd", exitOnError)
+
+	fmt.Println()
+	CheckAllPodsStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println("Info:  checking placement group status")
+	checkPgStatus(context, operatorNamespace, clusterNamespace, exitOnError)
+
+	fmt.Println()
+	fmt.Println("Info:  checking if at least one mgr pod is running")
+	checkMgrPodsStatusAndCounts(context, operatorNamespace, clusterNamespace, exitOnError)
+}
+
+func checkPodsOnNodes(context *k8sutil.Context, operatorNamespace, clusterNamespace, label string, exitOnError bool) {
+	var daemonType string
+	if strings.Contains(label, "osd") {
+		daemonType = "osd"
+	} else if strings.Contains(label, "mon") {
+		daemonType = "mon"
+	}
+
+	opts := metav1.ListOptions{LabelSelector: label}
+	podList, err := context.Clientset.CoreV1().Pods(clusterNamespace).List(ctx.TODO(), opts)
+	if err != nil {
+		fmt.Printf("\nfailed to list %s pods with label %s: %v\n", daemonType, opts.LabelSelector, err)
+	}
+
+	var nodeList = make(map[string]string)
+	for i := range podList.Items {
+		nodeName := podList.Items[i].Spec.NodeName
+		if _, okay := nodeList[nodeName]; !okay {
+			nodeList[nodeName] = podList.Items[i].Name
+		}
+	}
+
+	if len(nodeList) < 3 {
+		fmt.Printf("\nWarning: At least three %s pods should running on different nodes", daemonType)
+	}
+
+	for i := range podList.Items {
+		fmt.Println(podList.Items[i].Name, "\t", podList.Items[i].Status.Phase, "\t", podList.Items[i].Namespace, "\t", podList.Items[i].Spec.NodeName)
+	}
+}
+
+func checkMonQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	cephHealthDetails := exec.RunCommandInOperatorPod(context, "ceph", []string{"health", "detail"}, operatorNamespace, clusterNamespace, false)
+	if cephHealthDetails == "HEALTH_OK" {
+		fmt.Printf("\n%s", cephHealthDetails)
+	} else if cephHealthDetails == "HEALTH_WARN" {
+		fmt.Println("Warning: ", cephHealthDetails)
+	} else if cephHealthDetails == "HEALTH_ERR" {
+		fmt.Println("Error: ", cephHealthDetails)
+	}
+}
+
+func CheckAllPodsStatus(context *k8sutil.Context, operatorNamespace, clusterNamespace string, exitOnError bool) {
+	podListInOpNs, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
```

Comment:
for better readability, spell out var names. If they're long, then find simpler names.
```suggestion
	operatorNamespacePods, err := context.Clientset.CoreV1().Pods(operatorNamespace).List(ctx.TODO(), metav1.ListOptions{})
```